RELEASE.md

```diff
@@ -1,23 +1,233 @@
+
+## Branches and tags
+
+We maintain a branch per major release (after releasing 1.0 we may change to a branch per minor release - this is yet to be decided). This is done so that we are able to release an update for older major release (e.g. release 0.13.3 after 0.14.0 is released).
+Major version branches are named `branch-0.A.x`, where `A` is a number of the version. Example: branch for `0.14` will be `branch-0.14.x`.
+
+All new development lands on `main` branch, and may be backported by maintainers to version branches.
+
+Each release of `scylla` crate has a corresponding git tag. Tags are named `vA.B.C`, so release `0.14.1` will have tag `v0.14.1`.
+Each such tag has a message of the form `Release 0.X.Y` (e.g. `Release 0.14.0`).
+
+Below is a simplified view of how this looks.
+Diagrams created using [mermaid.js](https://mermaid.js.org/). If you view this file on GitHub it should be rendered correctly.
+You can also view it using [mermaid.live](https://mermaid.live)
+
+```mermaid
+---
+title: Rust Driver git branches
+---
+gitGraph
+   commit id: "Feature A"
+   commit id: "Release 0.13" tag: "v0.13.0"
+   branch branch-0.13.x
+   checkout branch-0.13.x
+   commit id: "Backport fix C"
+   commit id: "Release 0.13.1" tag: "v0.13.1"
+   commit id: "Backport fix D"
+   checkout main
+   commit id: "Feature B"
+   commit id: "Fix C"
+   commit id: "Release 0.14" tag: "v0.14.0"
+   branch branch-0.14.x
+   checkout branch-0.14.x
+   commit id: "Backport fix D " %% Space at the end because mermaid doesn't 
+   %% allow duplicate labels and only displays one of the commits with duplicate labels...
+   commit id: "Release 0.14.1" tag: "v0.14.1"
+   checkout main
+   commit id: "Fix D"
+```
+
+## Version bump commits
+
+For simplicity the graph in previous section suggested that version bumps happen in one commit.
+In reality we bump each crate version in a separate commit.
+
+For an example, see four last commits of this PR: https://github.com/scylladb/scylla-rust-driver/pull/1040/commits
+
+Commits that update `scylla-proxy` / `scylla-macros` / `scylla-cql` change:
+- `version` field in relevant `Cargo.toml`
+- Version number in crates that depend on given crate (so a commit bumping `scylla-macros` will also change version of 
+`scylla-macros` in `scylla` and `scylla-cql` crates).
+- Version number of the crate in `Cargo.lock.msrv`
+
+Additionally the last commit (which bumps `scylla` crate version) changes:
+- `docs/pyproject.toml` and `docs/source/conf.py`, as described in "Documentation" section.
+- Version number in `docs/source/quickstart/create-project.md`.
+
+
+## Backporting
+
+If you want to backport a whole PR, you should chery-pick the merge commit. You will need `-m` flag for that - read about it in the manual.
+
+You can also backport just a specific commit / commits of bigger PR, also using cherry-pick.
+
+Please use `-x` flag for cherry-pick to record information about original commit.
```

Comment:
Real life examples would be appreciated here. I suggest:
- one example for cherry-picking a whole PR,
- one for a single-commit PR,
- one for a single commit from a multi-commit PR.